foodsaving/users/models.py

```diff
@@ -217,3 +219,25 @@ def has_perm(self, perm, obj=None):
     def has_module_perms(self, app_label):
         # temporarily only allow access for admins
         return self.is_superuser
+
+    def give_trust_by(self, given_by):
+        Trust.objects.update_or_create(user=self, given_by=given_by, defaults={'valid_from': timezone.now()})
+
+    def revoke_trust_by(self, given_by):
+        Trust.objects.get(user=self, given_by=given_by).delete()
+
+
+class TrustManager(Manager):
+    def valid(self):
+        return self.filter(valid_from__gte=timezone.now() - relativedelta(days=settings.TRUST_VALID_TIME))
+
+
+class Trust(BaseModel):
+    objects = TrustManager()
+    user = ForeignKey(User, on_delete=models.CASCADE, related_name='trusts')
```

Comment:
This could use a UniqueTogether constraint on `(user, given_by`) to keep the database in good shape. Have a look at the Feedback model and serializer.